pin-code.js

```diff
@@ -62,10 +63,27 @@ class CodePin extends Component {
 
   handleEdit(number, id) {
     let newCode = this.state.code.slice();
-    newCode[id] = number;
+
+    // Detecting if the entire code has been pasted or autofilled into
+    // the first field.
+    const hasAutofilled = number.length > 1 &&
+                          number.length === this.props.number &&
```

Comment:
is it possible that autofill code doesn't match the required digit number ?

In this case : for exemple if user autofill "124" but the code is "1245".  The app will crash.
 
I suggest to remove `number.length === this.props.number` condition on line **70** and remplace `|| hasAutofilled` code on line **86** by `|| number.length === this.props.number`.

Do you think it's a good solution ?

Reply:
I'm not sure how I feel about that - I'm assuming that the user is configuring the component correctly, so setting `number` correctly. 

Whatever solution they are using to generate the code will generate codes of a predictable length, and if it doesn't, then that is a larger UX problem since you may not be rendering the correct number of TextField's for manual input.

Reply:
Also just a clarification on current behavior - If I paste in "123" but I have `number` set to 4 then the app will not crash. Then the result will be 1 in the first field - Line 82 takes care of that.

The reason for that change is that previously you would end up with `["123", "",  "", ""]` which seemed bad! 

Looking at it the morning after I can see a few other ways we may want to refactor. I'll push an update shortly.

Reply:
I didn't look at it right. 

I understand better now. thanks for your time !

Reply:
I can not easily test autofill. So I trust you on that.

For me your solution it's ok for me. 

If you have time to handle bad length autofill to not catch only the first number, It'll be perfect. But otherwise it's possible to improve it later.